### ISSUE TITLE
xDnsARecord Test-TargetResource returns incorrect result

### ISSUE THREAD

**Issue by iainbrighton** [PowerShell/xDnsServer#6]
The first configuration completes as the resource doesn't exist. A second pass, however, fails as the resource attempts to recreate the A record (presumably as the Test-TargetResource function returns $false when it shouldn't?).
```
VERBOSE: Perform operation 'Invoke CimMethod' with following parameters, ''methodName' = SendConfigurationApply,'className' = MSFT_DSC
LocalConfigurationManager,'namespaceName' = root/Microsoft/Windows/DesiredStateConfiguration'.
VERBOSE: An LCM method call arrived from computer DC1 with user sid S-1-5-21-369477980-2918075026-776283088-500.
VERBOSE: [DC1]: LCM:  [ Start  Set      ]
VERBOSE: [DC1]: LCM:  [ Start  Resource ]  [[xDnsARecord]test_virutalengine_co_uk]
VERBOSE: [DC1]: LCM:  [ Start  Test     ]  [[xDnsARecord]test_virutalengine_co_uk]
VERBOSE: [DC1]:                            [[xDnsARecord]test_virutalengine_co_uk] Testing for DNS test in virtualengine.co.uk
VERBOSE: [DC1]:                            [[xDnsARecord]test_virutalengine_co_uk] Looking up DNS record for test in virtualengine.co.
uk
VERBOSE: [DC1]: LCM:  [ End    Test     ]  [[xDnsARecord]test_virutalengine_co_uk]  in 0.0160 seconds.
VERBOSE: [DC1]: LCM:  [ Start  Set      ]  [[xDnsARecord]test_virutalengine_co_uk]
VERBOSE: [DC1]:                            [[xDnsARecord]test_virutalengine_co_uk] Creating for DNS 1.2.3.4 in virtualengine.co.uk
Failed to create resource record test in zone virtualengine.co.uk on server localhost.
    + CategoryInfo          : ResourceExists: (test:) [], CimException
    + FullyQualifiedErrorId : WIN32 9711,Add-DnsServerResourceRecordA
    + PSComputerName        : localhost
 
VERBOSE: [DC1]: LCM:  [ End    Set      ]  [[xDnsARecord]test_virutalengine_co_uk]  in 0.0150 seconds.
The PowerShell DSC resource MSFT_xDnsARecord threw one or more non-terminating errors while running the Set-TargetResource 
functionality. These errors are logged to the ETW channel called Microsoft-Windows-DSC/Operational. Refer to this channel for more 
details.
    + CategoryInfo          : InvalidOperation: (:) [], CimException
    + FullyQualifiedErrorId : NonTerminatingErrorFromProvider
    + PSComputerName        : localhost
```

**Comment by iainbrighton** [comment]
OK - it looks like the .ToString() method on the '$record.RecordData' property returns 'DnsServerResourceRecordA' and not the IP Address. This then fails in the comparison in the Test-TargetResource function.

If I change the following in the Get-TargetResource function to the following, it works:
```
return @{
        Name = $record.HostName
        Zone = $Zone
        Target = $record.RecordData.IPv4Address.IPAddressToString
    }
```
Needs a little more testing as I've only tested this on a 2012R2 box. However, the Set-TargetResource method is no longer called:
```
VERBOSE: [DC1]: LCM:  [ Start  Set      ]
VERBOSE: [DC1]: LCM:  [ Start  Resource ]  [[xDnsARecord]test_virutalengine_co_uk]
VERBOSE: [DC1]: LCM:  [ Start  Test     ]  [[xDnsARecord]test_virutalengine_co_uk]
VERBOSE: [DC1]:                            [[xDnsARecord]test_virutalengine_co_uk] Testing for DNS test in virtualengine.co.uk
VERBOSE: [DC1]:                            [[xDnsARecord]test_virutalengine_co_uk] Looking up DNS record for test in virtualengine.co.uk
VERBOSE: [DC1]: LCM:  [ End    Test     ]  [[xDnsARecord]test_virutalengine_co_uk]  in 1.0150 seconds.
VERBOSE: [DC1]: LCM:  [ Skip   Set      ]  [[xDnsARecord]test_virutalengine_co_uk]
VERBOSE: [DC1]: LCM:  [ End    Resource ]  [[xDnsARecord]test_virutalengine_co_uk]
VERBOSE: [DC1]: LCM:  [ End    Set      ]    in  1.7028 seconds.
```

**Comment by iainbrighton** [comment]
Initial fix here: https://github.com/iainbrighton/xDnsServer/tree/Issue6

**Comment by iainbrighton** [comment]
This was fixed in a commit 889e96d
